### PR TITLE
:up: sqlite3@5.1.2

### PR DESCRIPTION
**Pull request by bpasero**
Bring us up to date with https://github.com/TryGhost/node-sqlite3 with the following changes:
* drop `node-pre-gyp` so that the library is built natively
* add workflows for Electron testing
* preserve extended error codes
* some tweaks to native building

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,33 +45,18 @@
...
   "repository": {
     "type": "git",
-    "url": "https://github.com/TryGhost/node-sqlite3.git"
+    "url": "https://github.com/microsoft/vscode-node-sqlite3.git"
   },
   "dependencies": {
-    "@mapbox/node-pre-gyp": "^1.0.0",
-    "node-addon-api": "^4.2.0",
-    "tar": "^6.1.11"
+    "node-addon-api": "^4.2.0"
```
Comment (bpasero) [inline package.json:51]: @deepak1556 is this version ok to use? I noticed `5.0.0` is out.

File: binding.gyp [hunk 2]
```diff
@@ -6,15 +6,28 @@
...
-      "target_name": "<(module_name)",
+      "target_name": "vscode-sqlite3",
       "cflags!": [ "-fno-exceptions" ],
       "cflags_cc!": [ "-fno-exceptions" ],
       "xcode_settings": { "GCC_ENABLE_CPP_EXCEPTIONS": "YES",
         "CLANG_CXX_LIBRARY": "libc++",
-        "MACOSX_DEPLOYMENT_TARGET": "10.7",
+        # Target depends on
+        # https://chromium.googlesource.com/chromium/src/+/master/build/config/mac/mac_sdk.gni#22
+        "MACOSX_DEPLOYMENT_TARGET": "10.11",
```
Comment (bpasero) [inline binding.gyp:16]: @deepak1556 is this still reflecting our target in VS Code building?

File: binding.gyp [hunk 3]
```diff
@@ -6,15 +6,28 @@
...
       },
       "msvs_settings": {
-        "VCCLCompilerTool": { "ExceptionHandling": 1 },
+        "VCCLCompilerTool": {
```
Comment (bpasero) [inline binding.gyp:19]: @deepak1556 I assume these are still OK

File: binding.gyp [hunk 4]
```diff
@@ -49,18 +62,18 @@
         "src/node_sqlite3.cc",
         "src/statement.cc"
       ],
-      "defines": [ "NAPI_VERSION=<(napi_build_version)", "NAPI_DISABLE_CPP_EXCEPTIONS=1" ]
+      "defines": [ "NAPI_DISABLE_CPP_EXCEPTIONS=1", "NODE_API_SWALLOW_UNTHROWABLE_EXCEPTIONS" ]
```
Comment (bpasero) [inline binding.gyp:65]: @deepak1556 I wonder if `NAPI_VERSION=<(napi_build_version)` is needed, I noticed we do not set it in our fork

**Comment by bpasero** [comment]
@deepak1556 I left some comments in the code for a quick review. this is essentially taking over the changes we had, nothing really changed, but maybe some of these changes need an update. Thanks!

File: binding.gyp (hunk 2 above)
Comment (deepak1556) [reply binding.gyp:16]: Yup for Electron 19 , the deployment target is still `10.11`

File: binding.gyp (hunk 3 above)
Comment (deepak1556) [reply binding.gyp:19]: yup these are correct and needed for CodeQL compliance.

File: binding.gyp (hunk 4 above)
Comment (deepak1556) [reply binding.gyp:65]: The value is only needed by `node-pre-gyp` https://github.com/mapbox/node-pre-gyp/blob/master/README.md#the-napi_build_version-value which we don't use, so I removed it.

File: package.json (hunk 1 above)
Comment (deepak1556) [reply package.json:51]: We can update to the latest `5.0.0`. The current locked version was based on whatever was latest at that time.

**Review by deepak1556 — APPROVED** [review]
LGTM, not merging incase you want to try updating `node-addon-api` in this PR.

File: package.json (hunk 1 above)
Comment (bpasero) [reply package.json:51]: Oh nevermind, upstream is using the same version!

**Pull request merged**